### ISSUE TITLE
Why do we switch to 12 columns at $breakpoint-medium?

### ISSUE THREAD

**Issue by lyubomir-popov** [canonical-web-and-design/vanilla-framework#3209]
## Pattern to amend

Grid: Is there a reason the switch to 12 columns should happen at such a narrow width? It makes it very difficult to use 4 column layouts. At 72rem, or even 1440px as we use it in maas, 4 columns could be very useful. Shrink the width to 772px, and text becomes unreadable. I wonder if there are any negatives to switching at a more reasonable width like $breakpoint-large, which is 1036px. 

@bartaz or @anthonydillon do you know why it is currently so low??

**Comment by anthonydillon** [comment]
Switch to 12 column? Do you mean when the grid stacks?

Are you referring to maas.io or maas-ui?

**Comment by lyubomir-popov** [comment]
We change from 4 to 6 columns, then  from 6 to 12. I just wonder why the switch to 12 in vanilla (not any specific site) happens at 772 instead of 1036 px, i.e. at $breakpoint-medium instead of $breakpoint-large. The problem this creates is that you have to make sure any layout looks acceptable when the width of a column is 1/4 of 772, which eliminates 4-column layouts in most cases.

**Comment by anthonydillon** [comment]
I'm unsure but can look into it. I think examples would help. Maybe someting for the working group?

**Comment by bartaz** [comment]
I don't think we can easily do this without affecting possibly a lot of places on the websites. Most notably - ubuntu.com takeovers.

![Screenshot 2021-03-25 at 09.16.49.png](https://images.zenhubusercontent.com/593a660851aa2c30578102bb/7081637d-ecff-4028-8d4f-2efdfcb376fe)

It's caused by the fact that switching the grid columns is often accompanied by show/hide utils being used to show/hide content (like takeover images).
Currently for example takeovers use `u-hide--small` to hide images when grid drops from 12 columns (on medium breakpoint).
If we change the grid breakpoint to `large` the images will be visible while the grid will already be at 6 cols.

This is not something we can change in Vanilla with CSS or settings. This will require changing or adding class names in HTML in possibly many places, and there is no easy way to identify these places. This would be a breaking change.

We can ask project to simply revert this change in their own settings if they find it breaking. But this probably means most of the project will have to do that. So, what's the point of changing something in Vanilla if we know products will have to override it.

In the long term it seems we would need show/hide utils that are tied to the layout of the grid, not to the screen size.
So, instead of using `u-hide--small` you would have `u-hide--6-col-grid`, to hide things when grid changes. This will allow to change the settings of the grid, without breaking the utilities.

**Comment by lyubomir-popov** [comment]
@bartaz nice exploration, thanks. Maybe something to add to the 3.0 list?
